include/socgraph.php

```diff
@@ -401,13 +412,47 @@ function poco_detect_server($profile) {
 
 	// Mastodon
 	if ($server_url == "") {
-		$red = preg_replace("=(https?://)(.*)/users/(.*)=ism", "$1$2", $profile);
-		if ($red != $profile) {
-			$server_url = $red;
+		$mastodon = preg_replace("=(https?://)(.*)/users/(.*)=ism", "$1$2", $profile);
+		if ($mastodon != $profile) {
+			$server_url = $mastodon;
 			$network = NETWORK_OSTATUS;
 		}
 	}
 
+	// Numeric OStatus variant
+	if ($server_url == "") {
+		$ostatus = preg_replace("=(https?://)(.*)/user/(.*)=ism", "$1$2", $profile);
+		if ($ostatus != $profile) {
+			$server_url = $ostatus;
+			$network = NETWORK_OSTATUS;
+		}
+	}
+
+	// Wild guess
+	if ($server_url == "") {
+		$base = preg_replace("=(https?://)(.*?)/(.*)=ism", "$1$2", $profile);
+		if (base != $profile) {
+			$server_url = $base;
+			$network = NETWORK_PHANTOM;
+		}
+	}
+
+	if ($server_url == "") {
+		return "";
+	}
+
+	$r = q("SELECT `id` FROM `gserver` WHERE `nurl` = '%s' AND `last_contact` > `last_failure`",
+		dbesc(normalise_link($server_url)));
+	if (dbm::is_result($r)) {
+		return $server_url;
+	}
+
+	// Fetch the host-meta to check if this really is a server
+	$serverret = z_fetch_url($server_url."/.well-known/host-meta");
+	if (!$serverret["success"]) {
```

Comment:
What about this: https://github.com/friendica/friendica/blob/develop/include/network.php#L76 ?